lib/command.rb

```diff
@@ -11,32 +11,78 @@ def initialize(message = nil, code = nil)
       super(message)
     end
   end
-
   class ExitError < CommandError; end
 
   def self.prepended(base)
-    base.prepend Command::SimpleCommand
-    base.include Command::ErrorHandling
+    base.extend ClassMethods
+  end
+
+  attr_reader :result
+
+  module ClassMethods
+    def call(*args, **kwargs)
+      new(*args, **kwargs).call
+    end
+
+    def extended(base)
+      base.i18n_scope = "errors.messages"
+    end
+    attr_accessor :i18n_scope
   end
 
   def call
     fail NotImplementedError unless defined?(super)
 
+    @called = true
     @result = super
-    @result
-  rescue ExitError => e
+    self
+  rescue ExitError => _e
+  end
+
+  def success?
+    called? && !failure?
   end
 
+  def failure?
+    called? && errors.any?
+  end
+
+  def errors
+    @errors ||= Command::Errors.new(source: self.class)
+  end
+
+  module LegacyErrorHandling
+    # Convenience/retrocompatibility aliases
+    def self.errors_legacy_alias(method, errors_method)
+      define_method method do |*args|
+        puts "/!\\ #{method} is deprecated, please use errors.#{errors_method} instead."
```

Comment:
```suggestion
        warn "/!\\ #{method} is deprecated, please use errors.#{errors_method} instead."
```
https://ruby-doc.org/core-2.6/Warning.html 😉